lib/IRGen/GenProto.cpp

```diff
@@ -1734,7 +1735,7 @@ void IRGenModule::emitSILWitnessTable(SILWitnessTable *wt) {
     if (ClassDecl *clas = dyn_cast<ClassDecl>(Nominal)) {
       if (clas->isForeign())
         getAddrOfForeignTypeMetadataCandidate(conformingType);
-    } else if (Nominal->hasClangNode()) {
+    } else if (isa<ClangModuleUnit>(Nominal->getModuleScopeContext())) {
```

Comment:
Should `hasClangNode` be replaced by a `isInClangModuleContext` method on ValueDecl? If it's the wrong thing to reach for for this kind of query, it seems like we should make the right thing at least as obvious.